proxy/proxy.go

```diff
@@ -882,7 +885,24 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 
 	p.metrics.IncCounter("outgoing." + req.Proto)
 	ctx.proxySpan.LogKV("http_roundtrip", StartEvent)
-	response, err := p.roundTripper.RoundTrip(req)
+
+	var response *http.Response
+	switch ctx.route.BackendType {
+	case eskip.FastCGIBackend:
+		connFactory := gofast.SimpleConnFactory("tcp", ctx.route.FcgiAddress)
+
+		rr := httptest.NewRecorder()
+
+		gofast.NewHandler(
+			gofast.NewFileEndpoint("index.php")(gofast.BasicSession),
+			gofast.SimpleClientFactory(connFactory, 0),
+		).ServeHTTP(rr, req)
+
+		response = rr.Result()
```

Comment:
Is it possible to use client connections pooling and then: https://godoc.org/github.com/yookoala/gofast#ClientFunc.Do instead of this hack (use of httptest.Recorder to get the response from an inline created handler )?

Reply:
You already highlighted as todo. :)

Reply:
Yeah I was wrestling with this. The `gofast` library uses a [ResponsePipe](https://godoc.org/github.com/yookoala/gofast#ResponsePipe) and Skipper wants a regular Response. Here my knowledge of Go limited me. How would you pull this off?

Reply:
Then have a type that makes it possible to get the http.Response from the ResponsePipe.
Best would be to use the pooled connections provided by https://godoc.org/github.com/yookoala/gofast#NewClientPool

Reply:
I decided to not do this. My first approach was to copy `httptest.NewRecorder` but what's the point of completely copying the httptest recorder? Can't we just use this? It gets the job done, and is battle tested.